tools/nightly_build_and_test_stats.groovy

```diff
@@ -147,18 +146,120 @@ def getLatestBinariesTag(String version) {
+    if ( testOutputRC != 0 ) {
+        echo "Warning: This URL's data could not be found, and is likely expired: ${url}"
+        return ""
+    }
+    return sh(returnStdout: true, script: "wget -q -O - ${url}")
+}
+
+// Return our best guess at the build TRSS IDs for the latest successful build+publish of a specific set of platforms for a specific build tag.
+// Takes a jdk major version, a tag, and an array of platform names (standard platform format).
+def getBuildIDsByPlatform(String trssUrl, String jdkVersion, String srcTag, Map platformsList) {
+    // First we gather a list of the latest pipelines for this jdkVersion.
+    echo "Gathering Build IDs by platform."
+    def jdkVersionMinusTheU = jdkVersion.endsWith("u") ? jdkVersion.substring(0, jdkVersion.length() - 1) : jdkVersion
+    def pipelineName = "open${jdkVersionMinusTheU}-pipeline"
+    def pipelines = callWgetSafely("${trssUrl}/api/getBuildHistory?buildName=${pipelineName}")
+    def pipelineJson = new JsonSlurper().parseText(pipelines)
+
+    if (pipelineJson.size() == 0) {
+        echo "WARNING: Cannot find pipelines for per-platform build job identification."
+        return
+    }
+
+    def platformConversionMap = getPlatformConversionMap()
+    Set platformKeys = platformsList.keySet()
+
+    // Then we iterate over the list of pipelines, seeking a pipeline that contains one of our platforms.
+    assert pipelineJson instanceof List
+
+    for (int i = 0 ; i < pipelineJson.size() ; i++ ) {
+        Map onePipeline = pipelineJson[i]
+
+        if (!onePipeline.toString().contains(srcTag.replaceAll("-beta",""))) {
+            continue
+        }
+
+        def pipelineBuilds = callWgetSafely("${trssUrl}/api/getChildBuilds?parentId=${onePipeline._id}")
+        def pipelineBuildsJson = new JsonSlurper().parseText(pipelineBuilds)
+
+        if (pipelineBuildsJson.size() == 0) {
+            continue
+        }
+
+        boolean pipelinePublishBool = false
+        def onePipelinePlatformsMap = [:]
+
+        // For each build within a given pipeline:
+        assert pipelineBuildsJson instanceof List
+        for (int j = 0 ; j < pipelineBuildsJson.size() ; j++ ) {
+            Map onePipelineBuild = pipelineBuildsJson[j]
+
+            // - Is this platform in our platform list?
+            for (int k = 0 ; k < platformKeys.size() ; k++ ) {
+                String onePlatformKey = platformKeys[k]
+                String onePlatformValue = platformsList[onePlatformKey]
+                if (!onePlatformValue.isEmpty()) {
+                    continue
+                }
+
+                if (onePipelineBuild.buildName.contains("${jdkVersion}-${platformConversionMap[onePlatformKey][0]}")) {
+                    // - Does the build job for one of our listed platforms contain a successful build job?
+                    if (onePipelineBuild.status.equals("Done") && (onePipelineBuild.buildResult.equals("UNSTABLE") || onePipelineBuild.buildResult.equals("SUCCESS"))) {
```

Comment:
although not ideal, checking buildResult is not quite accurate, as the build job maybe FAILURE if one of the sub-test jobs failed (eg.sanity.openjdk), so it will still have published the JDK artifact

In reality we need to check "Is there a JDK archive in the build artifacts", as if it has for an EA Beta build, then it will have been published...

Reply:
Ok, I've coded the changes for that [here](https://github.com/adoptium/ci-jenkins-pipelines/commit/0cd13e48b82de3cbc67b31496b05bab178b550c4), and I'm testing it [here](https://ci.adoptium.net/job/adam_temp_job_2/62/console).